### DISCUSSION TITLE
Is there a handy way to get the s3 url of the given key and bucket?

### DISCUSSION THREAD

**Discussion by kunalvarpe** [awspring/spring-cloud-aws#965] [Q&A] [answered]
I do not see any api which directly return the s3 URL. The way is to get the object using s3 template and then construct the URL.

**Comment by maciejwalkowiak** [comment, marked as answer]
Perhaps `S3Resource#getURL` would do the job for you? https://docs.awspring.io/spring-cloud-aws/docs/3.0.3/apidocs/io/awspring/cloud/s3/S3Resource.html

**Reply by kunalvarpe** [reply]
@maciejwalkowiak Thank you, Stupid me, searching the method in respective clients 😮‍💨. Thank you for pointing that will do the job.

**Reply by maciejwalkowiak** [reply]
No worries it is quite hidden
